### DISCUSSION TITLE
Configuration of housekeeping tasks

### DISCUSSION THREAD

**Discussion by meks007** [blakeblackshear/frigate#4036] [Ask A Question]
Hi,

are there any configuration options to fine-tune the interval/time when housekeeping tasks run? I'm trying to set all maintenance tasks for my various systems to be between 1am and 3am in order to minimize hdd spin ups but it appears there are no such options in Frigate?

**Comment by NickM-27** [comment]
Frigate has `record -> expire_interval` which controls how often the cleanup runs, but there's currently no way to run it only at a specific time

**Comment by meks007** [comment]
thank you!

**Comment by meks007** [comment]
Nick, just for your information this appears to stricly apply to recordings only. I've set expire_interval to 1440 but it appers snapshots get deleted the moment they expire, like:

Oct 07 07:03:38 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1664255007.265365-bcdmz2.jpg
Oct 07 07:03:38 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1664255007.265365-bcdmz2-clean.png
Oct 07 07:48:38 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1664257625.699817-n18n7t.jpg
Oct 07 07:48:38 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1664257625.699817-n18n7t-clean.png
Oct 07 11:38:39 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_kamera3-1664271232.560247-t1j46z.jpg
Oct 07 11:38:39 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_kamera3-1664271232.560247-t1j46z-clean.png
Oct 07 12:08:40 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_kamera3-1664273115.954986-j89fbu.jpg
Oct 07 12:08:40 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_kamera3-1664273115.954986-j89fbu-clean.png

This leads to disks spinning almost the entire time.

**Reply by NickM-27** [reply]
Right, that was specific to record. I'll look into adding a similar option to snapshot

**Comment by meks007** [comment]
I'm sorry to bother you but would there be a chance to finally address that?
The way Frigate handles obsolete snapshots is a little stupid and prohibits disk spindown almost completely.

** Disk 1 **
Jun 05 12:52:12 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685098043.535792-myv5oj.jpg
Jun 05 12:52:12 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685098043.535792-myv5oj-clean.png
Jun 05 14:13:03 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685103097.522396-gksdlk.jpg
Jun 05 14:13:03 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685103097.522396-gksdlk-clean.png
Jun 05 14:23:11 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685103751.27212-squ7kr.jpg
Jun 05 14:23:11 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685103751.27212-squ7kr-clean.png
Jun 05 14:33:19 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685104331.984338-0rmvb4.jpg
Jun 05 14:33:19 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685104331.984338-0rmvb4-clean.png
Jun 05 15:08:41 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685106442.806799-p8o707.jpg
Jun 05 15:08:41 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685106442.806799-p8o707-clean.png
Jun 05 15:44:03 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685108429.522351-pcyp45.jpg
Jun 05 15:44:03 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685108429.522351-pcyp45-clean.png
Jun 05 16:19:26 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685110466.934391-3bo7g9.jpg
Jun 05 16:19:26 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685110466.934391-3bo7g9-clean.png
Jun 05 16:19:27 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685110493.975548-96b07w.jpg
Jun 05 16:19:27 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685110493.975548-96b07w-clean.png
Jun 05 16:24:32 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685111001.047011-p24zkd.jpg
Jun 05 16:24:32 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685111001.047011-p24zkd-clean.png
Jun 05 16:29:37 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685111328.632394-hlv1uf.jpg
Jun 05 16:29:37 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685111328.632394-hlv1uf-clean.png
Jun 05 16:49:51 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685112478.980907-1pj2rs.jpg
Jun 05 16:49:51 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685112478.980907-1pj2rs-clean.png
Jun 05 16:59:58 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685112902.184569-vxv9bn.jpg
Jun 05 16:59:58 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685112902.184569-vxv9bn-clean.png
Jun 05 17:10:13 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685113713.718577-kor8y1.jpg
Jun 05 17:10:13 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685113713.718577-kor8y1-clean.png
Jun 05 17:20:14 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685114242.616424-mv9vt8.jpg
Jun 05 17:20:14 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685114242.616424-mv9vt8-clean.png
Jun 05 17:40:32 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685115435.081533-idwdg6.jpg
Jun 05 17:40:32 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685115435.081533-idwdg6-clean.png
Jun 05 17:40:32 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685115438.91892-ybu2md.jpg
Jun 05 17:40:32 DELETE => /mnt/disk1/cctv/frigate/clips/cctv_doorbird-1685115438.91892-ybu2md-clean.png

**Reply by NickM-27** [reply]
I would suggest just not mounting snapshots on a disk, they don't take up much space anyway. That's what I do and haven't had any issues.

**Reply by meks007** [reply]
Right, I'll go that route then. Just wanted to check back since there was a PR that was closed as stale in March. Thanks.
